### PR TITLE
Update pascal voc12 example

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Updated pascal voc12 example to idist API
- refactored training files
  - single training
  - add exp_tracking module to dispatch calls to specific implementations: mlflow, plx, trains
  - updated docs/notes

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: examples/references/segmentation/pascal_voc2012/experiments/mlflow/conda.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ dependencies:
   - python=3.7
   - numpy
   - cudatoolkit=10.1.243=h6bb024c_0
-  - pytorch=1.3.1=py3.7_cuda10.1.243_cudnn7.6.3_0
-  - torchvision=0.4.2=py37_cu101
+  - pytorch=1.5.0=py3.7_cuda10.1.243_cudnn7.6.3_0
```
Comment (sdesrozis) [inline examples/references/segmentation/pascal_voc2012/experiments/mlflow/conda.yaml:9]: Does pin a specific build related to `amp` ? Or to ensure that `cuda` version is installed ? I don’t know if it’s a good idea to impose `cuda` version.

File: examples/references/segmentation/pascal_voc2012/configs/train/baseline_resnet101.py [hunk 2]
```diff
@@ -6,7 +6,8 @@
 import torch.nn as nn
 import torch.optim as optim
 import torch.optim.lr_scheduler as lrs
-import torch.distributed as dist
+
+import ignite.distributed as idist
```
Comment (sdesrozis) [inline examples/references/segmentation/pascal_voc2012/configs/train/baseline_resnet101.py:10]: Used?

File: examples/references/segmentation/pascal_voc2012/configs/train/baseline_resnet101_sbd.py [hunk 3]
```diff
@@ -6,7 +6,8 @@
 import torch.nn as nn
 import torch.optim as optim
 import torch.optim.lr_scheduler as lrs
-import torch.distributed as dist
+
+import ignite.distributed as idist
```
Comment (sdesrozis) [inline examples/references/segmentation/pascal_voc2012/configs/train/baseline_resnet101_sbd.py:10]: Used?

File: examples/references/segmentation/pascal_voc2012/README.md [hunk 4]
```diff
@@ -46,19 +47,34 @@ notebooks : jupyter notebooks to check specific parts from code modules
...
+- `common_training.py`, common training code used by above files -->
  
-Training scripts contain `run` method required by [py_config_runner](https://github.com/vfdev-5/py_config_runner) to 
-run a script with a configuration. Training logic is setup inside `training` method and configures a distributed trainer, 
-2 evaluators and various logging handlers to tensorboard, mlflow/polyaxon/Trains logger and tqdm.
-
+Training script contains `run` method required by [py_config_runner](https://github.com/vfdev-5/py_config_runner) to 
+run a script with a configuration. 
+
+The split between training script and configuration python file is the following.
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply examples/references/segmentation/pascal_voc2012/README.md:63]: I let you explore this option :)

File: examples/references/segmentation/pascal_voc2012/README.md [hunk 5]
```diff
@@ -46,19 +47,34 @@ notebooks : jupyter notebooks to check specific parts from code modules
...
+Configuration file being a python script defines necessary components for neural network training:
+- Dataflow: training/validation/train evaluation data loaders with custom data augmentations 
+- Model
+- Optimizer
+- Criterion
+- LR scheduler
+- other parameters: device, number of epochs, etc
+
+Training script uses these components to setup and run training and validation loops. By default, 
+processing group with "nccl" backend is initialized for distributed configuration (even for a single GPU).
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply examples/references/segmentation/pascal_voc2012/README.md:73]: IMO very limited interest of that. Would it be one of your use-cases ?

File: examples/references/segmentation/pascal_voc2012/experiments/mlflow/conda.yaml (hunk 1 above)
Comment (vfdev-5) [reply examples/references/segmentation/pascal_voc2012/experiments/mlflow/conda.yaml:9]: Good point about amp version and pytorch. The reason of fixing versions here is more or less historical and was related to BC incompatibilities between torchvision and pillow...

File: examples/references/segmentation/pascal_voc2012/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sdesrozis) [reply examples/references/segmentation/pascal_voc2012/README.md:73]: That does not matter for now 😉

**Pull request merged**